### PR TITLE
Rework the account/settings endpoint

### PR DESCRIPTION
**Pull request by Dorthu**
This is a proposal for how the new endpoints will look.  Please review
and comment here so I can implement.

### CODE REVIEW

File: src/data/endpoints/account.yaml [hunk 1]
```diff
@@ -36,23 +34,39 @@ endpoints:
...
+          curl: |
+            curl -h "Authorization: Bearer $TOKEN" \
+                https://$api_root/$version/account/settings
+      PUT:
+        description: >
+          Edits your account's defaults.
+        examples:
+          curl: |
+            curl -h "Content-Type: application/json" \
+                -h "Authorization: Bearer $TOKEn" \
```
Comment (yoshizzle) [inline src/data/endpoints/account.yaml]: capital N in TOKEN, please

File: src/data/objects/account.yaml [hunk 2]
```diff
@@ -16,10 +27,14 @@ schema:
     type: String
     description: City portion of this account's billing address.
     value: Philadelphia
-  company:
+  state:
     type: String
-    description: This account's company name.
-    value: My Company LLC
+    description: State (or province) portion of this account's billing address.
```
Comment (yoshizzle) [inline src/data/objects/account.yaml]: Might be helpful to explain in detail that if the customer is not in the US, please enter the province here. I know you have (or province) in parens, but I know from experience dealing with a lot of Asian folks - for example - that it's not entirely clear.

That one Thai customer account from two months ago comes to mind as well.

I guess I'm saying, let's add a little more detail in the description to be 100% perfectly clear.

File: src/data/objects/account.yaml [hunk 3]
```diff
@@ -16,10 +27,14 @@ schema:
     type: String
     description: City portion of this account's billing address.
```
Comment (yoshizzle) [inline src/data/objects/account.yaml]: do we need to say "portion"? Seems weird. Maybe:

The City for this account's billing address.

File: src/data/objects/account.yaml [hunk 4]
```diff
@@ -16,10 +27,14 @@ schema:
...
-  company:
+  state:
     type: String
-    description: This account's company name.
-    value: My Company LLC
+    description: State (or province) portion of this account's billing address.
+    value: PA
+  zip:
+    type: String
+    description: Zip Code portion of this account's billing address.
```
Comment (yoshizzle) [inline src/data/objects/account.yaml]: The Zip Code for this account's billing address.

File: src/data/endpoints/account.yaml [hunk 5]
```diff
@@ -5,28 +5,26 @@ description: >
...
+                https://$api_root/$version/account
       PUT:
         description: >
           Edits your account settings.
         examples:
           curl: |
-            curl -H "Content-Type: application/json" \
-                -H "Authorization: Bearer $TOKEN" \
+            curl -h "content-type: application/json" \
+                -h "authorization: bearer $token" \
```
Comment (yoshizzle) [inline src/data/endpoints/account.yaml]: Authorization
Bearer
$TOKEN

File: src/data/endpoints/account.yaml [hunk 6]
```diff
@@ -36,23 +34,39 @@ endpoints:
...
+      PUT:
+        description: >
+          Edits your account's defaults.
+        examples:
+          curl: |
+            curl -h "Content-Type: application/json" \
+                -h "Authorization: Bearer $TOKEn" \
+                -X PUT -d '{
+                  "network_helper": true,
                   "longview_subscription": "longview-10"
```
Comment (yoshizzle) [inline src/data/endpoints/account.yaml]: So there is nothing else to edit in account/settings now?

If there is, please add them. If not, then, uh... cool!

Comment (Dorthu) [reply src/data/endpoints/account.yaml]: with this proposal, there is not

File: src/data/objects/account.yaml [hunk 7]
```diff
@@ -14,12 +25,18 @@ schema:
...
-    description: City portion of this account's billing address.
+    description: The city for this account's billing address.
     value: Philadelphia
-  company:
+  state:
     type: String
-    description: This account's company name.
-    value: My Company LLC
+    description: >
+      State portion of this account's billing address, or providence if you
```
Comment (yoshizzle) [inline src/data/objects/account.yaml]: province, not providence

File: src/data/endpoints/account.yaml [hunk 8]
```diff
@@ -5,28 +5,26 @@ description: >
...
+            curl -h "Authorization: Bearer $TOKEN" \
+                https://$api_root/$version/account
       PUT:
         description: >
           Edits your account settings.
         examples:
           curl: |
-            curl -H "Content-Type: application/json" \
-                -H "Authorization: Bearer $TOKEN" \
+            curl -h "Content-Type: application/json" \
```
Comment (RobertDeRose) [inline src/data/endpoints/account.yaml]: why did these get changed to `-h` from `-H` when the man for curl says `-H` is correct?

File: src/data/endpoints/account.yaml [hunk 9]
```diff
@@ -36,23 +34,39 @@ endpoints:
...
+    description: >
+      Manage default settings for your account.
+    methods:
+      GET:
+        response: account_settings
+        description: >
+          Returns your current defaults.
+        examples:
+          curl: |
+            curl -h "Authorization: Bearer $TOKEN" \
```
Comment (RobertDeRose) [inline src/data/endpoints/account.yaml]: same here

File: src/data/endpoints/account.yaml [hunk 10]
```diff
@@ -36,23 +34,39 @@ endpoints:
...
+                https://$api_root/$version/account/settings
+      PUT:
+        description: >
+          Edits your account's defaults.
+        examples:
+          curl: |
+            curl -h "Content-Type: application/json" \
```
Comment (RobertDeRose) [inline src/data/endpoints/account.yaml]: and here

File: src/data/endpoints/account.yaml (hunk 8 above)
Comment (Dorthu) [reply src/data/endpoints/account.yaml]: 😞  I think I hit `u` instead of `y` one time and didn't notice

File: src/data/objects/account.yaml (hunk 7 above)
Comment (RobertDeRose) [reply src/data/objects/account.yaml]: why you hating on Rhode Island?

File: src/data/endpoints/account.yaml (hunk 9 above)
Comment (RobertDeRose) [inline src/data/endpoints/account.yaml]: `-H` here too

File: src/data/endpoints/account.yaml (hunk 10 above)
Comment (RobertDeRose) [inline src/data/endpoints/account.yaml]: `-H` here as well

File: src/data/objects/account.yaml (hunk 7 above)
Comment (yoshizzle) [reply src/data/objects/account.yaml]: This is still not fixed

File: src/data/objects/account.yaml [hunk 11]
```diff
@@ -1,9 +1,20 @@
-# vim: set et ts=2 sw=2 tw=80 :
 name: Account
-prefix: account/settings
+prefix: account
 description: >
-    A representation of account settings.
+    Your account contact and billing information.
 schema:
+  first_name:
```
Comment (RobertDeRose) [inline src/data/objects/account.yaml]: Missing PUT output

File: src/data/endpoints/account.yaml [hunk 12]
```diff
@@ -36,23 +34,39 @@ endpoints:
...
+                }' \
+                https://$api_root/$version/account
+  /account/settings:
+    group: Settings
+    type: resource
+    description: >
+      Manage default settings for your account.
+    methods:
+      GET:
+        response: account_settings
```
Comment (RobertDeRose) [inline src/data/endpoints/account.yaml]: Should probably include something like `is_managed`

Comment (Dorthu) [reply src/data/endpoints/account.yaml]: Done

**Pull request merged**